primitives/tests/api.rs

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Test the API surface of `primitives`.
+//!
+//! The point of these tests are to check the API surface as opposed to test the API functionality.
+//!
+//! ref: <https://rust-lang.github.io/api-guidelines/about.html>
```

Comment:
Maybe it's more worthwhile to call out what section(s) of the api-guidelines this is addressing.

Reply:
There are comments in the code eg C-DEBUG